solution/index.js

```diff
@@ -0,0 +1,548 @@
+                newList = "in-progress"
+                replaceList(taskId, task, newList)
+                break;
+            case "Digit3":
+                newList = "done"
+                replaceList(taskId, task, newList)
+                break;
+        }
+        document.taskElement = null;
+        document.removeEventListener("keydown", keydownEvent);
+    }
+};
+//----Search by query - Search input----\\
+//Task content scan and match to query
+function searchHandler(e){
+    //Get input value and convert to lower case
+    const inputValue = e.target.value.toLowerCase();
+    //Get list of tasks from the main container
+    const tasks = mainContainer.getElementsByClassName("task");
+    //Convert list to array and create a taskName variable for each task 
+    Array.from(tasks).forEach(function(task){
+        const taskName = task.firstChild.textContent;
+        //Compare between taskName (lower case) and the inputValue
+        if(taskName.toLowerCase().indexOf(inputValue) != -1){
+            //If isn't indexOf - block
+            task.style.display = 'block';
+        }else{
+            //If indexOf - none
+            task.style.display = 'none';
+        }
+    });
+};
+
+
+
+//--------API---------\\
+loadButton.addEventListener("click", loadHandler);
+updateButton.addEventListener("click", updateHandler);
+
+function resetTasksDom() {
+    storedTasks = getStoredTasks();
+    for(let task of storedTasks.todo){
+        let taskElement = document.getElementById(task.id)
+        if(taskElement != null){
+            taskElement.remove();
+        }
+    }
+    for(let task of storedTasks["in-progress"]){
+        let taskElement = document.getElementById(task.id)
+        if(taskElement != null){
+            taskElement.remove();
+        }
+    }
+    for(let task of storedTasks.done){
+        let taskElement = document.getElementById(task.id)
+        if(taskElement != null){
+            taskElement.remove();
+        }
+    }
+};
+
+const apiURL = 'https://json-bins.herokuapp.com/bin/614b360bb2e21e6812f6e73b';
+//Load event handler - receives the data in the API bin, saves it in the local storage and generates the DOM
+async function loadHandler(e){
+    let response = await fetch(apiURL, {
+        method: "GET",
+        headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+    });
+    let results = await response.json();
+    if(response.status < 400 || response.status === 418){
+        //Resets the DOM
+        resetTasksDom();
+        //Appends the new local stored values to Tasks
+        localStorage.setItem("tasks", JSON.stringify(results.tasks))
+        Tasks = getStoredTasks();
+        //Regenerate the DOM
+        generateTasksDom();
+    }else{
+        alert("Request failiure");
+    }
+};
+
+//Update event handler - Receives the data in the local storage and fetches it to the API
+async function updateHandler(e){
```

Comment:
Use better and more descriptive naming conventions (for example: saveTasks, loadTasks)